src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeNameParserHelpers.cs

```diff
@@ -102,29 +102,54 @@ static int GetUnescapedOffset(ReadOnlySpan<char> input, int startOffset)
             static bool NeedsEscaping(char c) => c is '[' or ']' or '&' or '*' or ',' or '+' or EscapeCharacter;
         }
 
-        internal static ReadOnlySpan<char> GetName(ReadOnlySpan<char> fullName)
+        internal static ReadOnlySpan<char> GetNamespace(ReadOnlySpan<char> fullName)
         {
-            // The two-value form of MemoryExtensions.LastIndexOfAny does not suffer
-            // from the behavior mentioned in the comment at the top of GetFullTypeNameLength.
-            // It always takes O(m * i) worst-case time and is safe to use here.
+            // Detect if a nested type name has a namespace (e.g. "a.b+c") and throw if it does.
```

Comment:
I meant that the Namespace property should unconditionally throw for nested types, irrespective of whether the nested type name has a dot in it.

I am sorry that I was not clear about the suggested behavior.

Reply:
Updated to always throw in the `Namespace` property, and to just slice after the last dot in `GetNamespace`.